pubsub/iterator.go

```diff
@@ -96,6 +101,10 @@ type messageIterator struct {
 	eoMu                      sync.RWMutex
 	enableExactlyOnceDelivery bool
 	sendNewAckDeadline        bool
+
+	enableTracing bool
+	// This maps trace parent spans to ackIDs, used for otel tracing.
```

Comment:
Given we don't get declare type via explicit initialization, maybe clarify the types for keys and values stored here?  Comment seems to implies the opposite of actual use (span->ackid vs ackid->span).

Also, do we need to worry about growth of this map, or are we guaranteed to always process the LoadAndDelete for all ackids?  Is any kind of LRU handling needed to terminate abandoned spans?

Reply:
Fixed the comment, and good catch. It maps ackIDs (strings) -> spans (trace.Span)

This map is periodically cleaned up similar to the pending `keepAliveDeadlines` map that was used before. There's a background thread that occasionally purges expired messages.